### PR TITLE
Updating hadoop version for standalone

### PR DESCRIPTION
**Pull request by CuriousVini**
Upgrading Hadoop version for standalone to 2.8.0.

Standalone ITN: 
image

ITM: https://builds.cask.co/browse/CDAP-ITM-122
Build: https://builds.cask.co/browse/CDAP-RUT993-1

### CODE REVIEW

**Comment by sreevatsanraman** [comment]
Have we tested on a cluster? Having a compile dependency might not work on all distros. Marking it as on-hold until it is tested.

**Comment by CuriousVini** [comment]
Build and ITN is green.

File: cdap-standalone/pom.xml [hunk 1]
```diff
@@ -151,7 +251,83 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-yarn-server-common</artifactId>
+      <version>2.8.0</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-yarn-api</artifactId>
+      <version>2.8.0</version>
```
Comment (chtyim) [inline cdap-standalone/pom.xml]: Use maven property to define the version.

File: cdap-standalone/pom.xml [hunk 2]
```diff
@@ -100,6 +148,52 @@
       <groupId>co.cask.cdap</groupId>
       <artifactId>cdap-app-fabric</artifactId>
       <version>${project.version}</version>
+      <exclusions>
```
Comment (chtyim) [inline cdap-standalone/pom.xml]: Since you added hadoop as direct dependency, exclusion is not needed.

**Comment by chtyim** [comment]
So this change means something can work in SDK but not in cluster?

**Comment by anew** [comment]
why 2.8? Woiuld it make more sense to support in SDK whatever we support in distributed as the minimum version? Now, can it happen that someone tests an app in SDK and it works, but when she deploys the same app in their cluster, it fails?

**Comment by anew** [comment]
what's the JIRA and the use case for this PR?

**Comment by CuriousVini** [comment]
@anew yea we have [ADLS plugin](https://github.com/hydrator/file-batch-source-plugins/blob/master/adls-plugins/src/main/java/co.cask.hydrator.plugin.source/ADLSBatchSource.java#L79) which requires hadoop 2.8.

**Comment by CuriousVini** [comment]
@anew @chtyim We have a product requirement of making ADLS plugin work with sdk cloud sandbox as per [CDAP-11599](https://issues.cask.co/browse/CDAP-11599). ADLS requires  [hadoop 2.8](https://github.com/apache/hadoop/blob/release-2.8.0-RC2/hadoop-tools/hadoop-azure-datalake/src/main/java/org/apache/hadoop/fs/adl/AdlFileSystem.java).

We discussed couple of options:

- Upgrade hadoop accross cdap repo, but it requires twill changes
- To change hadoop versions only for sdk.

I am still open for more suggestions.

**Comment by anew** [comment]
This is a reason why SDK != Azure sandbox. We should not upgrade the hadoop version for everybody, just because there is one particular environment that needs it.

**Comment by anew** [comment]
OK, after discussing in person I understand this better. LGTM

**Comment by CuriousVini** [comment]
Thanks @anew for your inputs! As we discussed, I have created following jiras: https://issues.cask.co/browse/CDAP-11608, https://issues.cask.co/browse/CDAP-11600.

**Comment by CuriousVini** [comment]
Build is green. ITN passed on cluster and standalone.

**Pull request merged**